ooniapi/services/ooniprobe/src/ooniprobe/routers/v1/probe_services.py

```diff
@@ -202,3 +233,362 @@ def probe_update_post(probe_update: ProbeUpdate) -> ProbeUpdateResponse:
+    request: Request,
+    response: Response,
+    check_in: CheckIn,
+    cc_reader: CCReaderDep,
+    asn_reader: ASNReaderDep,
+    clickhouse: ClickhouseDep,
+    settings: SettingsDep,
+) -> CheckInResponse:
+
+    # TODO: Implement throttling
+    run_type = check_in.run_type
+    charging = check_in.charging
+    probe_cc = check_in.probe_cc.upper()
+    probe_asn = check_in.probe_asn
+    software_name = check_in.software_name
+    software_version = check_in.software_version
+
+    resp, probe_cc, asn_i = probe_geoip(
+        request,
+        probe_cc,
+        probe_asn,
+        cc_reader,
+        asn_reader,
+    )
+
+    # On run_type=manual preserve the old behavior: test the whole list
+    # On timed runs test few URLs, especially when on battery
+    if run_type == "manual":
+        url_limit = 9999  # same as prio.py
+    elif charging:
+        url_limit = 100
+    else:
+        url_limit = 20
+
+    try:
+        charging = bool(charging)
+    except Exception:
+        log.error(
+            f"check-in params: {url_limit} '{probe_cc}' '{charging}' '{run_type}' '{software_name}' '{software_version}'"
+        )
+
+    if check_in.web_connectivity is not None:
+        catcodes = check_in.web_connectivity.get("category_codes", [])
+        if isinstance(catcodes, str):
+            category_codes = catcodes.split(",")
+        else:
+            category_codes = catcodes
+
+    else:
+        category_codes = []
+
+    for c in category_codes:
+        assert c.isalpha()
+
+    try:
+        test_items, _1, _2 = generate_test_list(
+            clickhouse, probe_cc, category_codes, asn_i, url_limit, False
+        )
+    except Exception as e:
+        log.error(e, exc_info=True)
+        # TODO: use same failover as prio.py:list_test_urls
+        # failover_generate_test_list runs without any database interaction
+        # test_items = failover_generate_test_list(country_code, category_codes, limit)
+        test_items = []
+
+    Metrics.CHECK_IN_TEST_LIST_COUNT.set(len(test_items))
+
+    conf: Dict[str, Any] = dict(
+        features={
+            # TODO(https://github.com/ooni/probe-cli/pull/1522): we disable torsf until we have
+            # addressed the issue with the fast.ly based rendezvous method being broken
+            "torsf_enabled": False,
+            "vanilla_tor_enabled": True,
+        }
+    )
+
+    # set webconnectivity_0.5 feature flag for some probes
+    # Temporarily disabled while we work towards deploying this in prod:
+    # https://github.com/ooni/probe/issues/2674
+    #
+    # octect = extract_probe_ipaddr_octect(1, 0)
+    # if octect in (34, 239):
+    #    conf["features"]["webconnectivity_0.5"] = True
+
+    conf["test_helpers"] = generate_test_helpers_conf()
+
+    resp["tests"] = {
+        "web_connectivity": {"urls": test_items},
+    }
+    resp["conf"] = conf
+    resp["utc_time"] = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    test_names = (
```

Comment:
Eventually we probably should place this list inside of the `common` so we don't have to make sure it's in sync in the multiple places where we use it.